### ISSUE TITLE
Download results from Flow Information page

### ISSUE THREAD

**Issue by jawilson0502** [google/grr#517]
It would be greatly beneficial to my workflow to be able to download the results of a flow or hunt from either the "Flow Information" page, or hunts "Overview" page. 

I find that I don't want to parse the results via the results page, but would rather download it and parse it off line. It can take a noticeable amount of time to load the results, which makes it more inconvenient to download it from there.

**Comment by hollowing** [comment]
What about using the API and use the get hunt result or get flow result API calls? Would that suite your needs?

**Comment by hollowing** [comment]
Could the slow behaviour relates to #466 on your side?

**Comment by jawilson0502** [comment]
I do not believe #466 to this issue - It's quite snappy when I visit my hunts page, as long as I don't have thousands upon thousands of results. I'm also using the HTTP Datastore. Usually around the 10,000 flow results I start seeing a lag (Which I think is almost to be expected TBH)

The API tends to be incredibly cumbersome, particularly in a production environment. (See https://groups.google.com/forum/#!topic/grr-users/7FSKBwlDd4w as to what was recommended to me). This is not a scalable option for multiple investigators.

**Comment by mbushkov** [comment]
AdminUI loads only first 50 results when "Results" tab is shown. The fact that the total number of results affects the latency of a query to fetch first 50 results hints at the datastore slowness problems, meaning that #466 may be related to this issue. Still, we should make the UI more usable in cases when the datastore is too slow.

I think what we want is one of these solutions:
1) Have export functionality on the "information" tab.
2) Use a special "export" tab, so that one doesn't have to go to "results" to do the export.
3) Show the "Export as CSV" button on "results" tab right away, without waiting for actual results to be loaded and shown.

I lean towards 3), because it would be a cleanest solution from the UI standpoint: everything related to results will be on "results" tab. It will also probably the simplest fix to make. Jessica, what do you think?

**Comment by jawilson0502** [comment]
I do wonder, if we are talking about a slow database (and as a disclaimer, I haven't dived into the code for this), will this essentially cause 2 requests for the same data to be made to the database?  The first request for loading on the results page, and the second request for generating the CSV? If so, I think option 3 would not be a great solution, as it will most likely overburden an already slow database. In this scenario, I think option 1 would be best, though I would be just as fine with option 2.

If what I've described above is not how any of this works, then I'm fine with option 3.

**Comment by mbushkov** [comment]
Jessica, thanks for an update. You're right: 3) will result in effectively 2 database requests. It's hard to say if this would overburden the database - to make this assumption we have to basically understand whether database slowness is IO- or CPU- bound. But it's a valid concern nevertheless. We'll probably proceed with 2) then.
